### PR TITLE
[DLP] Implemented Inspect Datastore data and Send findings to SCC

### PR DESCRIPTION
**Pull request by bhavika-dhanwani**
Implemented [Inspect Datastore entity and Send findings to SCC](https://cloud.google.com/dlp/docs/sending-results-to-scc.md#datastore-data)

Checklist
 - [X] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [X] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [X] These samples need a new/updated **env vars** in testing projects set to pass (Datastore Namespace and Datastore Kind for configuring Datastore data)
- [X] **Tests** pass:   `mvn clean verify` **required**
- [X] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample 
- [X] Please **merge** this PR for me once it is approved

### CODE REVIEW

**Review by soumya92 — APPROVED** [review]

File: dlp/snippets/src/main/java/dlp/snippets/InspectDatastoreSendToScc.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+          InspectJobConfig.newBuilder()
+              .setInspectConfig(inspectConfig)
+              .setStorageConfig(storageConfig)
+              .addActions(action)
+              .build();
+
+      // Construct the job creation request to be sent by the client.
+      CreateDlpJobRequest createDlpJobRequest =
+          CreateDlpJobRequest.newBuilder()
+              .setParent(LocationName.of(projectId, "global").toString())
```
Comment (Sita04) [inline dlp/snippets/src/main/java/dlp/snippets/InspectDatastoreSendToScc.java:114]: nit: Consider moving the "global" to a variable declared in main()

File: dlp/snippets/src/test/java/dlp/snippets/JobsTests.java [hunk 2]
```diff
@@ -125,4 +125,22 @@ public void testDeleteJobs() throws Exception {
...
+    assertThat(output).contains("Job created successfully");
+    String dlpJobName = output.split("Job created successfully: ")[1].split("\n")[0];
+
+    // Delete the created Dlp Job
+    DeleteDlpJobRequest deleteDlpJobRequest =
+        DeleteDlpJobRequest.newBuilder().setName(dlpJobName).build();
+
+    try (DlpServiceClient client = DlpServiceClient.create()) {
+      client.deleteDlpJob(deleteDlpJobRequest);
+    }
```
Comment (Sita04) [inline dlp/snippets/src/test/java/dlp/snippets/JobsTests.java:144]: If you plan to use `DlpServiceClient` to delete jobs in multiple test cases, consider moving the client initialization to @BeforeAll method and reuse the client across the test cases.

File: dlp/snippets/src/main/java/dlp/snippets/InspectDatastoreSendToScc.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+      StorageConfig storageConfig =
+          StorageConfig.newBuilder().setDatastoreOptions(datastoreOptions).build();
+
+      // Specify the type of info the inspection will look for.
+      List<InfoType> infoTypes =
+          Stream.of("EMAIL_ADDRESS", "PERSON_NAME", "LOCATION", "PHONE_NUMBER")
+              .map(it -> InfoType.newBuilder().setName(it).build())
+              .collect(Collectors.toList());
+
+      // The minimum likelihood required before returning a match:
```
Comment (Sita04) [inline dlp/snippets/src/main/java/dlp/snippets/InspectDatastoreSendToScc.java:82]:
```suggestion
      // The minimum likelihood required before returning a match.
```

**Review by Sita04 — APPROVED** [review]
Minor nits. Please consider fixing them before merge.

File: dlp/snippets/src/main/java/dlp/snippets/InspectDatastoreSendToScc.java (hunk 1 above)
Comment (bhavika-dhanwani) [reply dlp/snippets/src/main/java/dlp/snippets/InspectDatastoreSendToScc.java:114]: Generally, we are not declaring "global" in main() for all the code samples.
So if I make changes in current PR, we need to incorporate the changes for all the code sample thus I'm not making the requested changes.

**Comment by Sita04** [comment]
```
java.lang.NullPointerException
	at com.google.privacy.dlp.v2.PartitionId$Builder.setNamespaceId(PartitionId.java:709)
	at dlp.snippets.InspectDatastoreSendToScc.inspectDatastoreSendToScc(InspectDatastoreSendToScc.java:65)
	at dlp.snippets.JobsTests.testCreateDatastoreJobWithScc(JobsTests.java:137)
```
in `testCreateDatastoreJobWithScc`

**Comment by bhavika-dhanwani** [comment]
> ```
> java.lang.NullPointerException
> 	at com.google.privacy.dlp.v2.PartitionId$Builder.setNamespaceId(PartitionId.java:709)
> 	at dlp.snippets.InspectDatastoreSendToScc.inspectDatastoreSendToScc(InspectDatastoreSendToScc.java:65)
> 	at dlp.snippets.JobsTests.testCreateDatastoreJobWithScc(JobsTests.java:137)
> ```
> 
> in `testCreateDatastoreJobWithScc`

@Sita04 
Tests are failing as we need to set "DATASTORE_KIND" and "DATASTORE_NAMESPACE" environment variables. Could you please help us setting the same.

**Comment by Sita04** [comment]
@bhavika-dhanwani Kindly ping internally with the requirements.

File: dlp/snippets/src/test/java/dlp/snippets/TestBase.java [hunk 4]
```diff
@@ -38,6 +38,8 @@ abstract class TestBase {
   protected static final String SUBSCRIPTION_ID = System.getenv("PUB_SUB_SUBSCRIPTION");
   protected static final String DATASET_ID = System.getenv("BIGQUERY_DATASET");
   protected static final String TABLE_ID = System.getenv("BIGQUERY_TABLE");
+  protected static final String DATASTORE_NAMESPACE = System.getenv("DATASTORE_NAMESPACE");
+  protected static final String DATASTORE_KIND = System.getenv("DATASTORE_KIND");
```
Comment (annadata) [inline dlp/snippets/src/test/java/dlp/snippets/TestBase.java:42]: @bhavika-dhanwani Please update the variable names as per the reviewed doc.
“DLP_NAMESPACE_ID” and “DLP_DATASTORE_KIND”.

**Pull request merged**